src/fluree/db/nameservice/filesystem.cljc

```diff
@@ -1,85 +1,225 @@
-  (let [branch (if branch (name branch) "main")]
-    (go (file-address (str ledger-alias "/" branch "/head")))))
+  [base-address ledger-alias {:keys [branch] :as _opts}]
+  (when base-address
+    (str base-address ledger-alias)))
+
+(defn write-ns-record
+  [ns-record local-path alias]
+  (let [p-chan     (async/promise-chan) ;; return value
+        write-path (file-path local-path alias)
+        record-bs  (try* (json/stringify-UTF8 ns-record)
+                         (catch* e
+                                 (log/error "Error json-encoding nameservice record for ledger: " alias
+                                            "with exception: " (ex-message e)
+                                            "Original record where error occurred: " ns-record)
+                                 ;; return exception, don't throw.. will check for it below
+                                 (ex-info (str "Exception encoding file nameservice file for ledger: " alias)
+                                          {:status 500 :error :db/invalid-commit})))]
+    (if (util/exception? record-bs)
+      (async/put! p-chan record-bs)
+      #?(:clj  (async/thread
+                 (try
+                   (fs/write-file write-path record-bs)
+                   (async/put! p-chan write-path)
+                   (catch Exception e
+                     (log/error (str "Exception writing file nameservice file for ledger: " alias
+                                     "with exception: " (ex-message e))
+                                e)
+                     (async/put! p-chan (ex-info (str "Exception writing file nameservice file for ledger: " alias
+                                                      " with exception: " (ex-message e))
+                                                 {:status 500 :error :db/invalid-commit}
+                                                 e)))))
+         :cljs (try*
+                 (fs/write-file write-path record-bs)
+                 (async/put! p-chan write-path)
+                 (catch* e (async/put! p-chan e)))))
+    p-chan))
 
 (defn push!
-  "Just write to a different directory?"
-  [local-path {commit-address :address
-               nameservices   :ns}]
-  (let [my-ns-iri   (some #(when (re-matches #"^fluree:file:.+" (:id %)) (:id %)) nameservices)
-        commit-path (address-path commit-address)
-        head-path   (address-path my-ns-iri)
-        write-path  (str local-path "/" head-path)
-
-        work        (fn [complete]
-                      (log/debug (str "Updating head at " write-path " to " commit-path "."))
-                      (fs/write-file write-path (bytes/string->UTF8 commit-path))
-                      (complete (file-address head-path)))]
-    #?(:clj  (let [p (promise)]
-               (future (work (partial deliver p)))
-               p)
-       :cljs (js/Promise. (fn [resolve reject] (work resolve))))))
+  "The file nameservice will eventually hold metadata including various
```

Comment:
I generally try to keep notes for future work as comments instead of docstrings, it makes it easier to tell what the function _actually_ does as opposed to what it will do.

Reply:
Fixed in forthcoming PR to apply this changes to latest main